### PR TITLE
Added getters for JsonPatch and Operations

### PR DESCRIPTION
**Pull request by GavinF17**
Hoping this gets picked up & released due to recent activity by @Capstan :)

- Added JsonPatch.getOperations()
- Added field getters for operations
  - DualPathOperation.getFrom()
  - JsonPatchOperation.getOp() and JsonPatchOperation.getPath()
  - PathValueOperation.getValue()

Personal use case is needing to determine which class to patch when using subtypes, and the type info being held in a field which may be modified by a replace operation.

Solves #33 
Supercedes #45

### CODE REVIEW

**Comment by Spellmaker** [comment]
I'm also highly interested in this PR getting merged, getters would be really useful for examining the operations (in my use-case: checking for certain paths to discourage changes to some part of the json document)

**Comment by serac** [comment]
My use case is similar to @Spellmaker but notably different. We have a JSON model that maps to database models where only some fields are mutable via our REST API. We need to be able to example a patch request in order to validate that the JSON paths are mutable according to business policy. While we could do this with raw JSON and then parse it into a `JsonPatch` object after validation, it would be far more convenient to work with the parsed version where the operation and path are easily accessible on a `JsonPatch` object.

As far as this PR, it seems innocuous as `operations` is final and wrapped in an immutable collection, so there's no risk of clients mutating the internal state by exposing it via a getter. I hope you'll consider this enhancement.

File: src/main/java/com/github/fge/jsonpatch/DualPathOperation.java [hunk 1]
```diff
@@ -72,6 +72,10 @@ public final void serializeWithType(final JsonGenerator jgen,
         serialize(jgen, provider);
     }
 
+    public JsonPointer getFrom() {
```
Comment (Capstan) [inline src/main/java/com/github/fge/jsonpatch/DualPathOperation.java:75]: Why not `final`? Here and elsewhere.

File: src/main/java/com/github/fge/jsonpatch/JsonPatchOperation.java [hunk 2]
```diff
@@ -94,6 +94,14 @@ protected JsonPatchOperation(final String op, final JsonPointer path)
     public abstract JsonNode apply(final JsonNode node)
         throws JsonPatchException;
 
+    public String getOp() {
+        return op;
```
Comment (Capstan) [inline src/main/java/com/github/fge/jsonpatch/JsonPatchOperation.java:98]: Should this protectively copy the string?

File: src/main/java/com/github/fge/jsonpatch/PathValueOperation.java [hunk 3]
```diff
@@ -73,6 +73,10 @@ public final void serializeWithType(final JsonGenerator jgen,
         serialize(jgen, provider);
     }
 
+    public JsonNode getValue() {
+        return value;
```
Comment (Capstan) [inline src/main/java/com/github/fge/jsonpatch/PathValueOperation.java:77]: This is the one where if `JsonNode` is mutable, then callers can change the value out from under the `Operation`. Should we not protectively `deepCopy` here?

File: src/main/java/com/github/fge/jsonpatch/DualPathOperation.java (hunk 1 above)
Comment (GavinF17) [reply src/main/java/com/github/fge/jsonpatch/DualPathOperation.java]: Should have had this in, implemented now

File: src/main/java/com/github/fge/jsonpatch/PathValueOperation.java (hunk 3 above)
Comment (GavinF17) [reply src/main/java/com/github/fge/jsonpatch/PathValueOperation.java]: Added `deepCopy` as suggested, I didn't realise `JsonNode` is mutable so good spot!

File: src/main/java/com/github/fge/jsonpatch/JsonPatchOperation.java (hunk 2 above)
Comment (GavinF17) [reply src/main/java/com/github/fge/jsonpatch/JsonPatchOperation.java:98]: I wouldn't have thought this is necessary as `String` is immutable, if I'm not thinking of something can you please explain?

**Comment by GavinF17** [comment]
> Looks good overall.

Thanks for the feedback, addressed all comments with [one outstanding](https://github.com/java-json-tools/json-patch/pull/79#discussion_r464405619)

File: src/main/java/com/github/fge/jsonpatch/JsonPatchOperation.java (hunk 2 above)
Comment (Capstan) [reply src/main/java/com/github/fge/jsonpatch/JsonPatchOperation.java:98]: Ah, you're right.

**Comment by elalex2021** [comment]
I am using v.1.13 (https://mvnrepository.com/artifact/com.github.java-json-tools/json-patch/1.13), but not able to get the patch operations using method getOperations(). You need to add it to public interface Patch. Could you please do that ASAP? Thanks.

**Comment by elalex2021** [comment]
I am using v.13 (https://mvnrepository.com/artifact/com.github.java-json-tools/json-patch/1.13), but still not able to use method getOperations(). 

image

**Comment by ceenlrnjim** [comment]
We are also interested in having the getOperations accessor and see that it isn't in 1.13 - are there plans for another release that will pick up this change?

**Pull request merged**